src/connector/src/sink/snowflake.rs

```diff
@@ -0,0 +1,279 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use risingwave_common::array::{Op, StreamChunk};
+use risingwave_common::buffer::Bitmap;
+use risingwave_common::catalog::Schema;
+use serde::Deserialize;
+use serde_json::Value;
+use serde_with::serde_as;
+use with_options::WithOptions;
+
+use super::encoder::{
+    JsonEncoder, RowEncoder, TimeHandlingMode, TimestampHandlingMode, TimestamptzHandlingMode,
+};
+use super::snowflake_connector::{SnowflakeHttpClient, SnowflakeS3Client};
+use super::writer::LogSinkerOf;
+use super::{SinkError, SinkParam};
+use crate::sink::writer::SinkWriterExt;
+use crate::sink::{DummySinkCommitCoordinator, Result, Sink, SinkWriter, SinkWriterParam};
+
+pub const SNOWFLAKE_SINK: &str = "snowflake";
+const MAX_BATCH_ROW_NUM: u32 = 1000;
+
+#[derive(Deserialize, Debug, Clone, WithOptions)]
+pub struct SnowflakeCommon {
+    /// The snowflake database used for sinking
+    #[serde(rename = "snowflake.database")]
+    pub database: String,
+
+    /// The corresponding schema where sink table exists
+    #[serde(rename = "snowflake.schema")]
+    pub schema: String,
```

Comment:
Can't we query the schema from snowflow?

Reply:
I haven't found a solution to query schema from snowflow, but the `schema` is just the schema name, which will be used to identify which pipe to sink to when sending `insertFiles` post request.

Reply:
Do we have the ability to query the schema from external REST API request? 🤔

Reply:
Not sure whether this (query the schema from external REST API request) can be done by the same credentials provided below.

Asking users to provide a schema also LGTM.